### ISSUE TITLE
@ionic/core is installed on non-ionic projects

### ISSUE THREAD

**Issue by chawes13** [LaunchPadLab/lp-subsection-generator#37]
Since `@ionic/core` is listed as an optional dependency, yarn will still attempt to install it, it just won't complain if the install fails (more details: https://classic.yarnpkg.com/lang/en/docs/dependency-types/#toc-optionaldependencies).

Refer to https://github.com/LaunchPadLab/sea-tow-client/pull/261#discussion_r1400600138 where this was noticed on a client project.

While I believe this package doesn't get included in the final bundle, we really shouldn't be installing dependencies that aren't relevant.

**Comment by chawes13** [comment]
@mwislek Any thoughts on how to address this here? My intuition is that maybe we need to separate out the packages. Maybe that would mean converting this to a monorepo with a shared /core?

**Comment by mwislek** [comment]
> @mwislek Any thoughts on how to address this issue? My intuition is that we might need to just separate out the packages. Maybe that would mean converting this to a monorepo with a shared /core?

@chawes13 Yeah, a monorepo with a shared core makes sense, I believe. I think the differences between the standard client and Ionic client usage are mostly around the templated files that are included for each subsection; the "core" is simply including the correct set of templates for each use and then performing project-specific/subsection name-specific updates. The goal would then be that the client template would include the client template-specific version of the tool (along with its dependencies) and the Ionic client template would include the Ionic client template-specific version of the tool (along with its different set of dependencies that include the Ionic core), is that what you're thinking?
